src/forwarders/DSROracleForwarderArbitrum.sol

```diff
@@ -5,10 +5,12 @@ import { ArbitrumForwarder } from 'xchain-helpers/forwarders/ArbitrumForwarder.s
 
 import { DSROracleForwarderBase } from './DSROracleForwarderBase.sol';
 
-contract DSROracleForwarderArbitrumOne is DSROracleForwarderBase {
+contract DSROracleForwarderArbitrum is DSROracleForwarderBase {
 
-    constructor(address _pot, address _l2Oracle) DSROracleForwarderBase(_pot, _l2Oracle) {
-        // Intentionally left blank
+    address public immutable l1CrossDomain;
```

Comment:
I'd put it in the `DSROracleForwarderBase`. It's the same for Arbitrum and Optimism.

Reply:
Not the same for Gnosis Chain though and we can't assume in general.

Reply:
There is always some kind of messenger contract on L1 that we send these messages to. The only difference now is that for Arbitrum and Optimism we use xchain-helpers forwarders that take L1 messenger address as an arg and for Gnosis we use a helper that has this address hardcoded. Imo if we move towards being chain agnostic, we should also have a generic Gnosis/AMB-style forwarder that takes an address in a constructor and then uses a different version of the forwarder, the one that takes the messenger address as an arg. It's probably gonna be a niche use-case or not a use-case at all to support non-Gnosis Gnosis-style bridging, but for the sake of unification I'd do it this way.

Reply:
I don't want to be stubborn here though, so maybe it can be done in a different PR, but imo putting messenger address in the implementation of the Arbitrum/Optimism forwarder seems like an unnecesary ode duplication, since we are already extracting.

Reply:
Usually yes, but there might not be. This pattern is very common, but I don't want to enforce it as a requirement since there may be an exception which breaks the rule. Having 1 address that's usually there is not a problem. This is duplication of 1-2 lines.

Reply:
Ok, let's keep it as is, I guess it's not a big deal now and it'll be easier if we'll have some weirder forwarders to support.